### PR TITLE
Support updating and deleting capture sources

### LINKED ISSUE #19: API should support updating and deleting capture sources
It would be helpful if we can update capture sources or delete them.
Ideally, code should verify that there are no video streams tied to that capture source.

### PR DESCRIPTION
**Pull request by scott97**
Closes #19 
Depends on videostreams being implemented first because we need to check there are none for a capture source before deleting

### CODE REVIEW

File: api/handlers/capturesource.go [hunk 1]
```diff
@@ -173,6 +182,14 @@ func GetCaptureSources(ctx *fiber.Ctx) error {
 	})
 }
 
+// intoGeoPoint converts a string containing two comma-separated values into a GeoPoint.
+func intoGeoPoint(location string) googlystore.GeoPoint {
+	parts := strings.Split(location, ",")
+	lat, _ := strconv.ParseFloat(parts[0], 64)
```
Comment (Saxon1) [inline api/handlers/capturesource.go:188]: no error capture ?

File: api/handlers/capturesource.go [hunk 2]
```diff
@@ -212,3 +224,72 @@ func CreateCaptureSource(ctx *fiber.Ctx) error {
...
+
+	// Check that capture source has no video streams associated with it.
+	// TODO: Implement this.
+
+	// Update data in the datastore.
+	store := ds_client.Get()
+	key := store.IDKey("CaptureSource", id)
+	var captureSource model.CaptureSource
+
+	if store.Update(context.Background(), key, func(e datastore.Entity) {
```
Comment (Saxon1) [inline api/handlers/capturesource.go:250]: Can the call to store.Update happen outside of the if condition?

**Review by Saxon1 — CHANGES_REQUESTED** [review]
just a couple of comments but mostly good

File: api/handlers/capturesource.go (hunk 1 above)
Comment (scott97) [reply api/handlers/capturesource.go]: This function now returns errors

File: api/handlers/capturesource.go (hunk 2 above)
Comment (scott97) [reply api/handlers/capturesource.go]: fixed

**Review by trekhopton — APPROVED** [review]

**Pull request merged**